internal/services/cdn/cdn_frontdoor_custom_domain_resource.go

```diff
@@ -297,26 +244,30 @@ func resourceCdnFrontDoorCustomDomainUpdate(d *pluginsdk.ResourceData, meta inte
 		tls := &cdn.AFDDomainHTTPSParameters{}
 		tlsSettings := d.Get("tls").([]interface{})
 		v := tlsSettings[0].(map[string]interface{})
-
-		secret := v["cdn_frontdoor_secret_id"].(string)
+		secretRaw := v["cdn_frontdoor_secret_id"].(string)
 
 		// NOTE: Cert type has to always be passed in the update else you will get a
 		// "AfdDomain.TlsSettings.CertificateType' is required but it was not set" error
 		tls.CertificateType = cdn.AfdCertificateType(v["certificate_type"].(string))
 
 		// NOTE: Secret always needs to be passed if it is defined else you will
 		// receive a 500 Internal Server Error
-		if secret != "" {
-			tls.Secret = expandResourceReference(secret)
+		if secretRaw != "" {
+			secret, err := parse.FrontDoorSecretIDInsensitively(secretRaw)
```

Comment:
this is coming from the config so will be being validated and should be being parsed case-sensitively:

```suggestion
			secret, err := parse.FrontDoorSecretID(secretRaw)
```